### PR TITLE
add timestamps to multiple choice proposals 

### LINKED ISSUE #409: Add timestamps to multiple choice proposals
mirrors the changes to cw-proposal-single

### PR DESCRIPTION
**Pull request by blue-note**
closes #409

### CODE REVIEW

File: contracts/cw-proposal-multiple/src/proposal.rs [hunk 1]
```diff
@@ -35,6 +35,10 @@ pub struct MultipleChoiceProposal {
     /// Information about the deposit that was sent as part of this
     /// proposal. None if no deposit.
     pub deposit_info: Option<CheckedDepositInfo>,
+    /// The timestamp at which this proposal was created.
+    pub created: Timestamp,
+    /// The timestamp at which this proposal's status last changed (e.g. Passed, Executed).
```
Comment (ezekiiel) [inline contracts/cw-proposal-multiple/src/proposal.rs:40]: Could we add a note here about how quorum expiration stuff makes this a little strange? Technically, it's possible for a proposal to become "passed" without this being updated so we don't want folks relying on this to be up to date.

Comment (ezekiiel) [reply contracts/cw-proposal-multiple/src/proposal.rs:40]: Hmm so looking at this more it looks like we currently don't update this when passed anywhere. Is that intentional? If not, should we be updating this in `execute_vote` if it causes the proposal to pass?

**Review by ezekiiel — COMMENTED** [review]
Looks good! One question about updating status on pass.

File: contracts/cw-proposal-multiple/src/proposal.rs (hunk 1 above)
Comment (janitachalam) [reply contracts/cw-proposal-multiple/src/proposal.rs:40]: so the timestamp is updated in `update_status` which is called by `execute_vote,` I think that should suffice?

Comment (ezekiiel) [reply contracts/cw-proposal-multiple/src/proposal.rs]: Oh I see. Yeah that's great!

**Review by ezekiiel — APPROVED** [review]
Awesome work! Thank you!

**Review by de-husk — COMMENTED** [review]
Looks like you need to regenerate the schema, since you updated the `contracts/cw-proposal-single` doc comments.

File: contracts/cw-proposal-multiple/src/contract.rs [hunk 2]
```diff
@@ -414,6 +419,8 @@ pub fn execute_close(
     };
 
     prop.status = Status::Closed;
+    // Update proposal's last updated timestamp.
+    prop.last_updated = env.block.time;
```
Comment (de-husk) [inline contracts/cw-proposal-multiple/src/contract.rs:423]: I see that we did the same thing in the single-choice PR, but Im just wondering why we didn't do this `prop.last_updated` update here as well?: https://github.com/DA0-DA0/dao-contracts/blob/main/contracts/cw-proposal-multiple/src/contract.rs#L750

**Review by de-husk — APPROVED** [review]

**Pull request merged**